src/DateTimePicker.jsx

```diff
@@ -118,20 +138,28 @@ class DateTimePicker extends React.Component {
 	* @param value String valid time string
 	*/
 	setTime(value) {
-		const newTime = new Date(value);
-		const datetime = new Date(this.state.datetime);
+		const datetime = this.parseNewTimeAsDate(value);
 
-		datetime.setTime(newTime.getTime());
 		this.setState({ datetime });
+
+		if (this.props.onChangeCallback) {
+			this.props.onChangeCallback(datetime);
+		}
```

Comment:
since the only difference between `setTime` and `setDateTime` is how the value of `datetime` is determined, `setTime` can just be

```js
setTime(value) {
  const datetime = this.parseNewTimeAsDate(value);
  return setDateTime(datetime);
}
```